go-controller/hybrid-overlay/pkg/controller/node.go

```diff
@@ -129,31 +130,35 @@ func NewNode(
 }
 
 // Run starts the controller
-func (n *Node) Run(stopCh <-chan struct{}) error {
+func (n *Node) Run(stopCh <-chan struct{}, done *sync.WaitGroup) {
 	defer utilruntime.HandleCrash()
 	klog.Info("Starting Hybrid Overlay Node Controller")
 
 	klog.Info("Starting workers")
+	done.Add(2)
 	go func() {
-		err := n.nodeEventHandler.Run(informer.DefaultNodeInformerThreadiness, stopCh)
+		err := n.nodeEventHandler.Run(informer.DefaultNodeInformerThreadiness, stopCh, done)
 		if err != nil {
 			klog.Error(err)
 		}
+		done.Done()
 	}()
 	go func() {
-		err := n.podEventHandler.Run(informer.DefaultInformerThreadiness, stopCh)
+		err := n.podEventHandler.Run(informer.DefaultInformerThreadiness, stopCh, done)
 		if err != nil {
 			klog.Error(err)
 		}
+		done.Done()
 	}()
 
-	go n.controller.RunFlowSync(stopCh)
+	done.Add(1)
```

Comment:
Either leave as is, or fold this in to the one above

Reply:
I'd like to leave as-is since it's not an event handler.